product_logistics_uom/hooks.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 ACSONE SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+import logging
+
+from odoo.tools import sql
+
+_logger = logging.getLogger(__name__)
+
+
+def pre_init_hook(env):  # pragma: nocover
+    """Recompute the volume and weight column on product and template
+    by converting the value from the uom defined on the product to the default uom
+    """
+    if sql.column_exists(env.cr, "product_template", "volume_uom_id"):
+        _logger.info("Recompute volume on product.product")
+        # get default m3 uom
+        env.cr.execute(
```

Comment:
```suggestion
        env.cr.execute(SQL("""
        SELECT res_id
            FROM ir_model_data
            WHERE module = 'uom' AND name = 'product_uom_cubic_meter'
            """)
            )
```
you can take the opportunity of the migration to use the new odoo.tools.SQL wrapper.
example: https://github.com/odoo/odoo/blob/69b404c7109ff689381f56520aad758424ec01aa/odoo/tools/sql.py#L306

Reply:
is need to update hook used SQL wrapper @xaviedoanhduy

Reply:
yes, u need. 
for example in version 16.0:
```
def column_exists(cr, tablename, columnname):
    """ Return whether the given column exists. """
    query = """ SELECT 1 FROM information_schema.columns
                WHERE table_name=%s AND column_name=%s """
    cr.execute(query, (tablename, columnname))
    return cr.rowcount
```
and in version 17.0, they are update it to used SQL wrapper:
```
def column_exists(cr, tablename, columnname):
    """ Return whether the given column exists. """
    cr.execute(SQL(
        """ SELECT 1 FROM information_schema.columns
            WHERE table_name=%s AND column_name=%s """,
        tablename, columnname,
    ))
    return cr.rowcount
```